electronics/Arduino/Portfile

```diff
@@ -2,16 +2,21 @@
 
 PortSystem          1.0
 PortGroup           github 1.0
-github.setup        arduino Arduino 1.8.8
+PortGroup           java 1.0
+
+github.setup        giansalvo Arduino 1.8.8-gians v
```

Comment:
I'm somewhat uncomfortable with using a forked repository. What did you fix in your fork that could not be attached as a patch, for example? Also, this changes the version of the port to `1.8.8-gians` which is probably not desired (one could use a suffix in github.setup to avoid this, but that's not the point).

The disadvantage is that:
* `livecheck` won't work, so we won't know when a new version gets out
* it will make it nearly impossible for any other contributor to reproduce your work when trying to update to a newer version

Reply:
I agree with you that this is not a perfect solution. In fact I got in touch with upstream developers and they explained that they removed some dependencies (JavaFX) that where making the macports build process fail, after releasing 1.8.8 version. I then forked their repository just to have a snapshot of their master trunk on which to calculate the checksums (maybe it's possible to do this on their repo... I don't know). I made no changes at all on the code. 
My idea is to get back to the official repository as soon as 1.8.9 version get released. Then I will implement livecheck.

Reply:
In that case use the shasum of the commit that works and set the version to 1.8.8 one line lower.

Reply:
What do you mean by "the relevant checksum"? do you mean this: cea22fbb6c10956429275db86dce98f86f92ff9ed8497bbf099f1a9e0df3d87e

Reply:
Not the tarball's checksum, but the commit id you get with `git log` or by browsing the github commits page. The latest one could be for example
```
github.setup arduino Arduino 2b11e94afe85888bd5ebd507de6c6e0220a49bf3
version 1.8.8
```
corresponding to https://github.com/arduino/Arduino/commit/2b11e94afe85888bd5ebd507de6c6e0220a49bf3.
(The sum might be shortened - you could use `2b11e94` for example, but some core developers prefer the full one. I like shorter ones, but we don't have a clearly written rule and honestly I don't care as long as it does the job.)

Reply:
Also, if you do this, you might want to leave a short note as to why you did it (like: build failure was fixed after 1.8.8 release). You would switch back to 1.8.9 as soon as that one gets released.

Reply:
Ok, that's exactly what I needed, thanks. I've already done some tests and it works. 
But before commiting just a couple of questions.
- Where should I leave the "short note" (inside a comment, in the description, ...)?
- Once built, the application recognize itself as "1.8.9" which is not correct and may be confusing for users. How should I deal with it?

Reply:
Just leave a short comment above the line with version number. Again, it's optional, just potentially helpful for the future in case anyone else stumbles upon the port.

I would not worry about the different version number reported by the software. You may call the version something like `1.8.8-git` if you want, to make it clear that it's not exactly the distributed version. Some developers add a timestamp, but that's only useful if you expect to keep updating from master branch, which is probably not the case here. If you call it `1.8.9-something`, then this would be bigger than 1.8.9 and you'll need to increase the epoch if you'll want to actually update to 1.8.9 later. This is just a temporary discrepancy anyway, don't overengineer the solution.

Reply:
Ok, done. How long do you think is going to take for the PR to be merged?